training/src/anemoi/training/train/scaling.py

```diff
@@ -0,0 +1,190 @@
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+import logging
+from abc import ABC
+from abc import abstractmethod
+
+import numpy as np
+from omegaconf import DictConfig
+
+from anemoi.models.data_indices.collection import IndexCollection
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BaseVariableLossScaler(ABC):
+    """Configurable method converting variable to loss scaling."""
+
+    def __init__(self, scaling_config: DictConfig, data_indices: IndexCollection) -> None:
+        """Initialise Scaler.
+
+        Parameters
+        ----------
+        scaling_config :
+        data_indices :
+
+        """
+        self.scaling_config = scaling_config
+        self.data_indices = data_indices
+        self.variable_groups = self.scaling_config.variable_groups
+        # turn dictionary around
+        self.group_variables = {}
+        for group, variables in self.variable_groups.items():
+            if isinstance(variables, str):
+                variables = [variables]
+            for variable in variables:
+                self.group_variables[variable] = group
+        self.default_group = self.scaling_config.variable_groups.default
+
+    @abstractmethod
+    def get_variable_scaling(self) -> np.ndarray: ...
+
+    def get_variable_group(self, variable_name: str) -> tuple[str, str, int]:
+        """Get the group of a variable.
+
+        Parameters
+        ----------
+        variable_name : str
+            Name of the variable.
+
+        Returns
+        -------
+        str
+            Group of the variable.
+
+        """
+        split = variable_name.split("_")
+        variable_level = None
+        if len(split) > 1 and split[-1].isdigit():
+            variable_level = int(split[-1])
+            variable_name = variable_name[: -len(split[-1]) - 1]
+        if variable_name in self.group_variables:
+            return self.group_variables[variable_name], variable_name, variable_level
+        return self.default_group, variable_name, variable_level
+
+
+class GeneralVariableLossScaler(BaseVariableLossScaler):
+    """General scaling of variables to loss scaling."""
+
+    def get_variable_scaling(self) -> np.ndarray:
+        variable_loss_scaling = (
+            np.ones((len(self.data_indices.internal_data.output.full),), dtype=np.float32) * self.scaling_config.default
+        )
+
+        for variable_name, idx in self.data_indices.internal_model.output.name_to_index.items():
+            _, variable_ref, _ = self.get_variable_group(variable_name)
+            # Apply variable scaling
+            variable_loss_scaling[idx] = self.scaling_config.get(
+                variable_ref,
+                1.0,
+            )
+            # TODO(all): do we want to allow scaling by variable ref and variable name?,
```

Comment:
I'm not sure what the difference between variable_ref and variable_name is here?

Reply:
`variable_ref` refers to this variable in the config weight list. 

So for q_10 the values would be:
- `variable_name=q_10`
- `variable_ref=q` (without the variable level)

I will add a few comments.